tests/test_resource.py

```diff
@@ -210,3 +214,134 @@ def a(self):
             "message": "Server got itself in trouble",
             "request_id": ANY,
         }
+
+
+ORIGIN_LOCALHOST = "http://localhost:3000"
+ORIGIN_EXAMPLE = "https://api.example.com"
+
+environ["CORS_ORIGIN"] = f"{ORIGIN_LOCALHOST},{ORIGIN_EXAMPLE}"
+
+
+class TestCORSResource:
+    def make_cors_handler(self, origins: List[str] = None, req_origin: str = None) -> CORSResource:
+        event = defaultdict(MagicMock())
+        event["headers"] = {"origin": req_origin} if req_origin is not None else {}
+        cors_handler = CORSResource(event, ["GET", "POST"], origins=origins)
+        return cors_handler
+
+    def test_cors_origin_headers_from_env_are_correct_1(self):
+        assert self.make_cors_handler().resp_headers_json[ALLOW_ORIGIN_HEADER] == ORIGIN_LOCALHOST
+
+    def test_cors_origin_headers_from_env_are_correct_2(self):
+        assert (
+            self.make_cors_handler(req_origin=ORIGIN_EXAMPLE).resp_headers_json[
+                ALLOW_ORIGIN_HEADER
+            ]
+            == ORIGIN_EXAMPLE
```

Comment:
Can we improve the formatting of all those tests? I think splitting into several lines will make them more readable 🙂